crates/net/eth-wire/src/types/transactions.rs

```diff
@@ -400,11 +400,11 @@ mod test {
                     Signature {
                         odd_y_parity: true,
                         r: U256::from_str(
-                            "35b7bfeb9ad9ece2cbafaaf8e202e706b4cfaeb233f46198f00b44d4a566a981",
+                            "0x35b7bfeb9ad9ece2cbafaaf8e202e706b4cfaeb233f46198f00b44d4a566a981",
                         )
                         .unwrap(),
                         s: U256::from_str(
-                            "612638fb29427ca33b9a3be2a0a561beecfe0269655be160d35e72d366a6a860",
+                            "0x612638fb29427ca33b9a3be2a0a561beecfe0269655be160d35e72d366a6a860",
```

Comment:
is the 0x prefix now required?

do the fromstr impls differ?

Reply:
Yes, but I do think that for a `uint` type it makes sense: https://github.com/recmo/uint/blob/main/src/string.rs#L170-L181

Reply:
if it's prefixed with 0x it does hex decoding, if it's not it'll do decimal, so imo it's nicer this way. no more `from_dec_str` footgun